### PR TITLE
Connect button contrast.  Progress ring when fetching repos.

### PR DESCRIPTION
**Pull request by dhoehna**
## Summary of the pull request
I didn't change the connect button to inherit from the accent style.  This caused the text color to clash with the accent color.

No UI indication is shown to the user when the UI is fetching the repos.  A progress ring now runs when repos are being fetched.

Also moved the bool to visibility converters to a more common location for other xaml files to user.
## References and relevant issues
#501 
#1099 
#1104 

## PR checklist
- [ X] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

Movie!
ProgressRingBetterContrastNotReDoRepoTool

Smaller progress ring!
SmallerProgressRing

### CODE REVIEW

**Comment by dhoehna** [comment]
Closes #501 
Closes #1099 
Closes #1104

**Comment by dhoehna** [comment]
Also.  Win UI 3 does not expose a thickness property on the progress ring.

File: tools/SetupFlow/DevHome.SetupFlow/Views/AddRepoDialog.xaml [hunk 1]
```diff
@@ -73,7 +70,8 @@
                     <ComboBox x:Name="AccountsComboBox" IsEnabled="{x:Bind AddRepoViewModel.IsAccountComboBoxEnabled, Mode=OneWay}" AutomationProperties.Name="ms-resource:///DevHome.SetupFlow/Resources/RepoTool_AccountComboBox_Description" Grid.Column="0" SelectionChanged="AccountsComboBox_SelectionChanged" ItemsSource="{x:Bind AddRepoViewModel.Accounts, Mode=OneWay}" />
                     <TextBox x:Name="FilterTextBox" Grid.Column="1" TextChanged="FilterTextBox_TextChanged" x:Uid="ms-resource:///DevHome.SetupFlow/Resources/RepoTool_FilterTextBox" />
                 </Grid>
-                <ListView x:Name="RepositoriesListView" SelectionMode="Multiple" Height="300" HorizontalAlignment="Stretch" SelectionChanged="RepositoriesListView_SelectionChanged" ItemsSource="{x:Bind AddRepoViewModel.Repositories, Mode=OneWay}" Margin="0, 12, 0, 12">
+                <ProgressRing Background="LightGray" Visibility="{x:Bind AddRepoViewModel.IsFetchingRepos, Mode=OneWay, Converter={StaticResource BoolToVisibilityConverter}}" Height="300" Width="300"/>
```
Comment (krschau) [inline tools/SetupFlow/DevHome.SetupFlow/Views/AddRepoDialog.xaml:73]: Do we want it so big? The other places we have a progress ring it's much smaller.

File: tools/SetupFlow/DevHome.SetupFlow/Converters/Converters.xaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿<ResourceDictionary
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:converters="using:CommunityToolkit.WinUI.UI.Converters">
+    <converters:BoolToVisibilityConverter x:Key="BoolToVisibilityConverter" TrueValue="Visible" FalseValue="Collapsed" />
+    <converters:BoolToVisibilityConverter x:Key="NegatedBoolToVisibilityConverter" TrueValue="Collapsed" FalseValue="Visible" />
```
Comment (asklar) [inline tools/SetupFlow/DevHome.SetupFlow/Converters/Converters.xaml:6]: don't need these

Comment (asklar) [inline tools/SetupFlow/DevHome.SetupFlow/Converters/Converters.xaml:6]: don't need these

Comment (dhoehna) [reply tools/SetupFlow/DevHome.SetupFlow/Converters/Converters.xaml:6]: Removed.

**Comment by dhoehna** [comment]
Because of reasons with the converters negating a bool with a converter and using that for visibility fails with the message "unable to cast object of type System.Boolean to Windows.Xaml.Ui.Visibility"

This means the converter to negate a boolean and turn it into a visibility has to still exist.

File: tools/SetupFlow/DevHome.SetupFlow/Views/AddRepoDialog.xaml [hunk 3]
```diff
@@ -117,7 +117,7 @@
                          https://github.com/microsoft/microsoft-ui-xaml/issues/2262 -->
                     <TextBlock>
                     <ToolTipService.ToolTip>
-                        <ToolTip x:Uid="ms-resource:///DevHome.SetupFlow/Resources/ClearCheckboxToBrowse" Visibility="{x:Bind FolderPickerViewModel.IsBrowseButtonEnabled, Mode=OneWay, Converter={StaticResource NegatedBoolToVisibilityConverter}}"/>
+                        <ToolTip x:Uid="ms-resource:///DevHome.SetupFlow/Resources/ClearCheckboxToBrowse" Visibility="{x:Bind FolderPickerViewModel.IsBrowseButtonEnabled, Mode=OneWay}"/>
```
Comment (krschau) [inline tools/SetupFlow/DevHome.SetupFlow/Views/AddRepoDialog.xaml:120]: Is this one not Negated anymore? Just making sure it wasn't an accident.

File: tools/SetupFlow/DevHome.SetupFlow/Views/AddRepoDialog.xaml [hunk 4]
```diff
@@ -25,9 +23,8 @@
             <ResourceDictionary.MergedDictionaries>
                 <ResourceDictionary Source="ms-appx:///DevHome.SetupFlow/Styles/SetupFlowStyles.xaml"/>
                 <ResourceDictionary>
-                    <converters:BoolToVisibilityConverter x:Key="BoolToVisibilityConverter" TrueValue="Visible" FalseValue="Collapsed"/>
-                    <converters:BoolToVisibilityConverter x:Key="NegatedBoolToVisibilityConverter" TrueValue="Collapsed" FalseValue="Visible"/>
                     <converters:BoolToObjectConverter x:Key="BoolToGlyphConverter" TrueValue="&#xF0BD;" FalseValue="&#xF03F;"/>
+                    <converters:BoolToVisibilityConverter x:Key="NegatedBoolToVisibilityConverter" TrueValue="Collapsed" FalseValue="Visible" />
```
Comment (dhoehna) [inline tools/SetupFlow/DevHome.SetupFlow/Views/AddRepoDialog.xaml:27]: This converter needs to stay.  The alternative is to include a 2nd boolean for the negated case, thus bloating the ViewModel and adding more logic to make sure both types stay is opposition.

File: tools/SetupFlow/DevHome.SetupFlow/Views/AddRepoDialog.xaml [hunk 5]
```diff
@@ -73,7 +70,10 @@
                     <ComboBox x:Name="AccountsComboBox" IsEnabled="{x:Bind AddRepoViewModel.IsAccountComboBoxEnabled, Mode=OneWay}" AutomationProperties.Name="ms-resource:///DevHome.SetupFlow/Resources/RepoTool_AccountComboBox_Description" Grid.Column="0" SelectionChanged="AccountsComboBox_SelectionChanged" ItemsSource="{x:Bind AddRepoViewModel.Accounts, Mode=OneWay}" />
                     <TextBox x:Name="FilterTextBox" Grid.Column="1" TextChanged="FilterTextBox_TextChanged" x:Uid="ms-resource:///DevHome.SetupFlow/Resources/RepoTool_FilterTextBox" />
                 </Grid>
-                <ListView x:Name="RepositoriesListView" SelectionMode="Multiple" Height="300" HorizontalAlignment="Stretch" SelectionChanged="RepositoriesListView_SelectionChanged" ItemsSource="{x:Bind AddRepoViewModel.Repositories, Mode=OneWay}" Margin="0, 12, 0, 12">
+                <Grid Visibility="{x:Bind AddRepoViewModel.IsFetchingRepos, Mode=OneWay}" Height="300" Width="300">
+                    <ProgressRing/>
```
Comment (krschau) [inline tools/SetupFlow/DevHome.SetupFlow/Views/AddRepoDialog.xaml:74]: Does this need to be bound to something?

**Review by krschau — CHANGES_REQUESTED** [review]

File: tools/SetupFlow/DevHome.SetupFlow/Views/AddRepoDialog.xaml (hunk 3 above)
Comment (dhoehna) [reply tools/SetupFlow/DevHome.SetupFlow/Views/AddRepoDialog.xaml:120]: You are correct.  Thank you.

File: tools/SetupFlow/DevHome.SetupFlow/Views/AddRepoDialog.xaml (hunk 5 above)
Comment (dhoehna) [reply tools/SetupFlow/DevHome.SetupFlow/Views/AddRepoDialog.xaml:74]: You mean the progress ring?  No.  It won't show if the grid is collapsed.

**Review by asklar — APPROVED** [review]

**Review by krschau — APPROVED** [review]

**Pull request merged**